### DISCUSSION TITLE
Why Keycloak team does not develop using the latest JDK version?

### DISCUSSION THREAD

**Discussion by Nefcanto** [keycloak/keycloak#25026] [Q&A] [answered]
Hi

I want to develop OTP for Keycloak and I started from [here](https://www.keycloak.org/getting-started/getting-started-zip) and it says to download JDK 17.

When I go to [JDK 17](https://jdk.java.net/17/) download page I see this message:

![Screenshot from 2023-11-26 15-50-00](https://github.com/keycloak/keycloak/assets/12782347/02d4e21e-4d41-494b-9870-022325317d0a)

In a bold emphasized tone it says:

> which do not include the most up to date security vulnerability fixes and are no longer recommended for use in production

So, what keeps Keycloak team from always staying on the most recent version of JDK?

**Comment by shawkins** [comment, marked as answer]
> When I go to [JDK 17](https://jdk.java.net/17/) download page I see this message:

The site is trying to push people to newer releases, however JDK 17 is a long term support release - https://en.wikipedia.org/wiki/Java_version_history and https://openjdk.org/projects/jdk/17/ - with support from vendors through 2024 and beyond.  In particular openjdk just released 17.0.9 with 17.0.10 scheduled early next year, and there will be support by Red Hat through 2027.

> So, what keeps Keycloak team from always staying on the most recent version of JDK?

Keycloak will align with Java LTS releases that have corresponding Quarkus LTS releases - https://quarkus.io/blog/lts-releases/
